File: app/src/main/java/com/example/vishistvarugeese/memory/Adapters/NotesAdapter.java
package com.example.vishistvarugeese.memory.Adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.vishistvarugeese.memory.R;
import com.example.vishistvarugeese.memory.Variables.Entity.Note;

import java.util.ArrayList;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {

    private ArrayList<Note> notesList = new ArrayList<>();
    private OnNoteListener onNoteListener;

    public NotesAdapter(OnNoteListener onNoteListener, ArrayList<Note> notesList){
        this.notesList = notesList;
        this.onNoteListener = onNoteListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.note_list_item, parent, false);
        return new ViewHolder(v, onNoteListener);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Note noteItem = notesList.get(position);
        holder.tv_noteDescription.setText(noteItem.getDescription());
        holder.tv_noteTitle.setText((noteItem.getTitle()));
        holder.tv_noteTimestamp.setText(noteItem.getTimestamp());

    }

    @Override
    public int getItemCount() {
        return notesList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
        private TextView tv_noteTitle;
        private TextView tv_noteDescription;
        private TextView tv_noteTimestamp;

        OnNoteListener mOnNoteListener;


        public ViewHolder(View itemView, OnNoteListener onNoteListener) {
            super(itemView);
            tv_noteTitle = (TextView) itemView.findViewById(R.id.tv_note_title);
            tv_noteDescription = (TextView) itemView.findViewById(R.id.tv_note_dialog_description);
            tv_noteTimestamp = (TextView) itemView.findViewById(R.id.tv_note_timestamp);
            mOnNoteListener = onNoteListener;
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            onNoteListener.onNoteClick(getAdapterPosition());
        }
    }

    public interface OnNoteListener{
        void onNoteClick(int position);
    }
}

File: app/src/main/java/com/example/vishistvarugeese/memory/Adapters/NotesAdapter.kt
package com.example.vishistvarugeese.memory.Adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.example.vishistvarugeese.memory.R
import com.example.vishistvarugeese.memory.Variables.Entity.Note
import java.util.*

class NotesAdapter(onNoteListener: OnNoteListener?, notesList: ArrayList<Note>) : RecyclerView.Adapter<NotesAdapter.ViewHolder>() {
    private val notesList = ArrayList<Note>()
    private val onNoteListener: OnNoteListener?
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.note_list_item, parent, false)
        return ViewHolder(v, onNoteListener!!)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val noteItem = notesList[position]
        holder.tv_noteDescription.text = noteItem.description
        holder.tv_noteTitle.text = noteItem.title
        holder.tv_noteTimestamp.text = noteItem.timestamp
    }

    override fun getItemCount(): Int {
        return notesList.size
    }

    inner class ViewHolder(itemView: View, onNoteListener: OnNoteListener) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        val tv_noteTitle: TextView
        val tv_noteDescription: TextView
        val tv_noteTimestamp: TextView
        var mOnNoteListener: OnNoteListener
        override fun onClick(v: View) {
            onNoteListener!!.onNoteClick(adapterPosition)
        }

        init {
            tv_noteTitle = itemView.findViewById<View>(R.id.tv_note_title) as TextView
            tv_noteDescription = itemView.findViewById<View>(R.id.tv_note_dialog_description) as TextView
            tv_noteTimestamp = itemView.findViewById<View>(R.id.tv_note_timestamp) as TextView
            mOnNoteListener = onNoteListener
            itemView.setOnClickListener(this)
        }
    }

    interface OnNoteListener {
        fun onNoteClick(position: Int)
    }

    init {
        this.notesList = notesList
        this.onNoteListener = onNoteListener
    }
}